File: 序号/167.两数之和II-输入有序数组.cpp
class Solution {
 public:
    vector<int> twoSum(vector<int>& numbers, int target) {
      int n=numbers.size();
        for(int i=0;i<n-1;i++){
          auto it = lower_bound(numbers.begin()+i+1,numbers.end(),target-numbers[i]);
          if(it==numbers.end()) continue;
          if(*it == target-numbers[i]) return {i+1,(int)(1+it-numbers.begin())};
        }
        return {};
    }
};

File: 序号/167. 两数之和 II - 输入有序数组.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        # d={}
        # for i,x in enumerate(numbers):
        #     if target-x in d:
        #         return [d[target-x]+1,i+1]
        #     d[x]=i

        n=len(numbers)
        l,r=0,n-1
        while l<r:
            s=numbers[l]+numbers[r]
            if s==target:
                return [l+1,r+1]
            if s<target:
                l+=1
            else:
                r-=1
        return []